consensus/src/round_manager.rs

```diff
@@ -759,46 +758,52 @@ impl RoundManager {
             (Author, Discriminant<VerifiedEvent>),
             (Author, VerifiedEvent),
         >,
+        close_rx: oneshot::Receiver<oneshot::Sender<()>>,
     ) {
         info!(epoch = self.epoch_state().epoch, "RoundManager started");
-        while let Some((peer_id, event)) = event_rx.next().await {
-            let result = match event {
-                VerifiedEvent::ProposalMsg(proposal_msg) => {
-                    monitor!(
-                        "process_proposal",
-                        self.process_proposal_msg(*proposal_msg).await
-                    )
+        let mut close_rx = close_rx.into_stream();
+        loop {
+            futures::select! {
```

Comment:
is there any fairness expected in the futures::select! ? 

i.e. if we got bunch of events in the event_rx, and we continue getting new ones, are we ever guaranteed to get an event for close_rx ?

Reply:
futures::select! guarantees some fairness by choosing one stream at random if multiple streams are ready - https://docs.rs/futures/0.3.0/futures/macro.select.html. However, in this case, the `close_rx` is outside of the select loop, so that is irrelevant in this context.

Reply:
it is not outside of select loop, if I am looking at it correctly?

Reply:
it's part of the select, but it relies on the fairness provided by the macro

Reply:
Any thoughts on moving this out of `select`? While the macro provides some fairness, it probably is better not to depend on that and move this out?

Reply:
how to move it out of select? this is the idiomatic way to receive messages from multiple channels